src/sentry/static/sentry/app/stores/guideStore.jsx

```diff
@@ -29,6 +26,18 @@ const GuideStore = Reflux.createStore({
     this.listenTo(GuideActions.unregisterAnchor, this.onUnregisterAnchor);
     this.listenTo(OrganizationsActions.setActive, this.onSetActiveOrganization);
     this.listenTo(OrganizationsActions.changeSlug, this.onChangeSlug);
+
+    window.addEventListener('hashchange', this.onHashChange, false);
+  },
+
+  onHashChange() {
+    if (
+      window.location.hash === '#assistant' &&
+      this.state.currentGuide &&
+      this.state.currentStep < 1
+    ) {
+      GuideActions.nextStep();
```

Comment:
Actions are meant to be messages from components to stores so we shouldn't be doing this. We should rather set a state variable indicating what the latest state of the hash is. Then use that in the logic that determines what to show (basically what your `forceShow` is doing currently)

Reply:
fixed